### PR TITLE
Embed version in the JavaScript client

### PR DESCRIPTION
**Pull request by anurse**
Related to https://github.com/aspnet/SignalR/issues/2130

The way this works for now is that it takes advantage of the fact that we update the `package.json` version from MSBuild during the packaging step. This happens **after** the build phase, so I added a script to the `prepack` phase in NPM that scans all files under `dist` (build outputs) and replaces a token with the actual version in `package.json`

### CODE REVIEW

File: clients/ts/build/embed-version.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+const SEARCH_STRING = "0.0.0-DEV_BUILD";
+function processFile(file) {
+    if (file.endsWith(".js") || file.endsWith(".ts")) {
+        if (verbose) {
+            console.log(`processing ${file}`);
+        }
+        let content = fs.readFileSync(file);
+        content = content.toString().replace(SEARCH_STRING, pkg.version);
```
Comment (BrennanConroy) [inline clients/ts/build/embed-version.js]: `pkg.version` just works?!

Comment (anurse) [reply clients/ts/build/embed-version.js]: Yep :)

File: clients/ts/signalr-protocol-msgpack/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+// Version token that will be replaced by the prepack command
+export const VERSION = "0.0.0-DEV_BUILD";
```
Comment (JamesNK) [inline clients/ts/signalr-protocol-msgpack/src/index.ts]: JS docs

File: clients/ts/signalr/src/index.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+// Version token that will be replaced by the prepack command
+export const VERSION = "0.0.0-DEV_BUILD";
```
Comment (JamesNK) [inline clients/ts/signalr/src/index.ts]: JS docs

**Comment by anurse** [comment]
> What can we do to unit test this?

We can't unit test this really, since the version isn't burned in until we package (which is after unit testing). I think for RTM this will just have to be a manual verification item.

**Comment by anurse** [comment]
https://github.com/aspnet/SignalR/issues/2256 filed for automation

**Comment by anurse** [comment]
:up: :date: Added a version number header to the UMD build. I didn't originally think it was necessary but it's useful for the minified version, where it's hard to see the version number. I used JSDoc "official" tags for both the version and the library title

File: clients/ts/signalr-protocol-msgpack/src/index.ts [hunk 4]
```diff
@@ -1,4 +1,8 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+// Version token that will be replaced by the prepack command
+/** The version of the SignalR client. */
```
Comment (JamesNK) [inline clients/ts/signalr-protocol-msgpack/src/index.ts]: Doc should say the version number is for the SignalR msgpack library

**Pull request merged**